src/GraphQL.Tests/Utilities/AstPrinterTests.cs

```diff
@@ -157,46 +85,50 @@ public void prints_node(INode node, string expected)
-            new object[] { new DecimalValue(0.00000000000000001m), "1E-17"},
-            new object[] { new BooleanValue(false), "false"},
-            new object[] { new BooleanValue(true), "true"},
-            new object[] { new EnumValue("TEST"), "TEST"},
+            new object[] { new GraphQLStringValue("test"), @"""test""" },
+            new object[] { new GraphQLStringValue("ab/cd"), @"""ab/cd""" },
+            new object[] { new GraphQLStringValue("ab\bcd"), @"""ab\bcd""" },
+            new object[] { new GraphQLStringValue("ab\fcd"), @"""ab\fcd""" },
+            new object[] { new GraphQLStringValue("ab\rcd"), @"""ab\rcd""" },
+            new object[] { new GraphQLStringValue("ab\ncd"), @"""ab\ncd""" },
+            new object[] { new GraphQLStringValue("ab\tcd"), @"""ab\tcd""" },
+            new object[] { new GraphQLStringValue("ab\\cd"), @"""ab\\cd""" },
+            new object[] { new GraphQLStringValue("ab\"cd"), @"""ab\""cd""" },
+            new object[] { new GraphQLStringValue("ab\u0019cd"), @"""ab\u0019cd""" },
+            new object[] { new GraphQLStringValue("\"abcd\""), @"""\""abcd\""""" },
+            new object[] { new GraphQLIntValue(int.MinValue), "-2147483648" },
+            new object[] { new GraphQLIntValue(0), "0" },
+            new object[] { new GraphQLIntValue(int.MaxValue), "2147483647" },
+            new object[] { new GraphQLIntValue(long.MinValue), "-9223372036854775808" },
+            new object[] { new GraphQLIntValue(0), "0" },
+            new object[] { new GraphQLIntValue(long.MaxValue), "9223372036854775807" },
+            new object[] { new GraphQLIntValue(new System.Numerics.BigInteger(decimal.MaxValue) * 1000 + 2), "79228162514264337593543950335002" },
+            new object[] { new GraphQLFloatValue(double.MinValue), "-1.79769313486232E+308" },
+            new object[] { new GraphQLFloatValue(double.Epsilon), "4.94065645841247E-324" },
+            new object[] { new GraphQLFloatValue(double.MaxValue), "1.79769313486232E+308" },
+            new object[] { new GraphQLFloatValue(0.00000001256), "1.256E-08" },
+            new object[] { new GraphQLFloatValue(12.56), "12.56" },
+            new object[] { new GraphQLFloatValue(3.33), "3.33" },
+            new object[] { new GraphQLFloatValue(1.0), "1" }, // double
+            new object[] { new GraphQLFloatValue((float)1.0), "1" },
+            new object[] { new GraphQLFloatValue((float)34), "34" },
+            new object[] { new GraphQLFloatValue(1.0000m), "1.0000" },
+            new object[] { new GraphQLFloatValue(1.0m), "1.0" },
+            new object[] { new GraphQLFloatValue(1m), "1" },
+            new object[] { new GraphQLFloatValue(decimal.MinValue), "-79228162514264337593543950335"},
+            new object[] { new GraphQLFloatValue(decimal.MaxValue), "79228162514264337593543950335"},
+            new object[] { new GraphQLFloatValue(0.00000000000000001), "1E-17"}, // double: G15 format
```

Comment:
@Shane32 Note changes here - `1E17` printed for double value, not for decimal. Decimal value printed as is.